src/test/java/core/basesyntax/SimpleCalculatorTest.java

```diff
@@ -0,0 +1,280 @@
+        double a = -15;
+        double b = -3;
+        double actual = calculator.calculate(a, b, '/');
+        double expected = 5;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void divisionWithNegativeAndPositive_OK() {
+        double a = 10;
+        double b = -2;
+        double actual = calculator.calculate(a, b, '/');
+        double expected = -5;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void divisionWithZero_OK() {
+        double a = 0;
+        double b = 10;
+        double actual = calculator.calculate(a, b, '/');
+        double expected = 0;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void divisionByZero_NotOK() {
+        double a = 10;
+        double b = 0;
+        assertThrows(ArithmeticException.class, () -> calculator.calculate(a, b, '/'));
+    }
+
+    @Test
+    void divisionWithMinAndMaxValues_NotOK() {
+        double a = Double.MAX_VALUE;
+        double b = Double.MIN_VALUE;
+        assertThrows(ArithmeticException.class, () -> calculator.calculate(a, b, '/'));
+    }
+
+    @Test
+    void raisingPositiveToPowerPositive_OK() {
+        double a = 5;
+        double b = 4;
+        double actual = calculator.calculate(a, b, '^');
+        double expected = 625;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void raisingPositiveToPowerNegative_OK() {
+        double a = 2;
+        double b = -3;
+        double actual = calculator.calculate(a, b, '^');
+        double expected = 0.125;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void raisingNegativeToPowerPositive_OK() {
+        double a = -5;
+        double b = 3;
+        double actual = calculator.calculate(a, b, '^');
+        double expected = -125;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void raisingNegativeToPowerNegative_OK() {
+        double a = -5;
+        double b = -3;
+        double actual = calculator.calculate(a, b, '^');
+        double expected = -0.008;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void raisingPositiveToPowerOfZero_OK() {
+        double a = 5;
+        double b = 0;
+        double actual = calculator.calculate(a, b, '^');
+        double expected = 1;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void raisingNegativeToPowerOfZero_OK() {
+        double a = -5;
+        double b = 0;
+        double actual = calculator.calculate(a, b, '^');
+        double expected = 1;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void raisingZeroToPower_OK() {
+        double a = 0;
+        double b = 5;
+        double actual = calculator.calculate(a, b, '^');
+        double expected = 0;
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void illegalOperation_NotOK() {
+        double a = 20;
+        double b = 20;
+        assertThrows(RuntimeException.class, () -> calculator.calculate(a, b, 'h'));
```

Comment:
```suggestion
        double b = 20;
        char invalidOperand = 'h';
        assertThrows(RuntimeException.class, () -> calculator.calculate(a, b, invalidOperand));
```
краще явно вказати чому певний чар тут викликає ексепшин